lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -151,68 +198,121 @@
         if 'volume_size' not in volume:
-            module.fail_json(msg='Size must be specified when creating a new volume or modifying the root volume')
+            module.fail_json(msg='Size must be specified when creating a new ' +
+                             'volume or modifying the root volume')
     if 'snapshot' in volume:
-        if 'device_type' in volume and volume.get('device_type') == 'io1' and 'iops' not in volume:
+        if 'device_type' in volume and volume.get(
+                'device_type') == 'io1' and 'iops' not in volume:
             module.fail_json(msg='io1 volumes must have an iops value set')
     if 'ephemeral' in volume:
         if 'snapshot' in volume:
             module.fail_json(msg='Cannot set both ephemeral and snapshot')
-    return BlockDeviceType(snapshot_id=volume.get('snapshot'),
-                           ephemeral_name=volume.get('ephemeral'),
-                           size=volume.get('volume_size'),
-                           volume_type=volume.get('device_type'),
-                           delete_on_termination=volume.get('delete_on_termination', False),
-                           iops=volume.get('iops'),
-                           encrypted=volume.get('encrypted',None))
+
+    return_object = {}
+
+    if 'ephemeral' in volume:
+        return_object['VirtualName'] = volume.get('ephemeral')
+
+    if 'device_name' in volume:
+        return_object['DeviceName'] = volume.get('device_name')
+
+    if 'no_device' is volume:
+        return_object['NoDevice'] = volume.get('no_device')
+
+    if any(key in volume for key in ['snapshot', 'volume_size',
+           'volume_type', 'delete_on_termination', 'ips', 'encrypted']):
+        return_object['Ebs'] = {}
+
+    if 'snapshot' in volume:
+        return_object['Ebs']['SnapshotId'] = volume.get('snapshot')
+
+    if 'volume_size' in volume:
+        return_object['Ebs']['VolumeSize'] = volume.get('volume_size')
+
+    if 'volume_type' in volume:
+        return_object['Ebs']['VolumeType'] = volume.get('volume_type')
+
+    if 'delete_on_termination' in volume:
+        return_object['Ebs']['DeleteOnTermination'] = volume.get(
+            'delete_on_termination', False)
+
+    if 'iops' in volume:
+        return_object['Ebs']['Iops'] = volume.get('iops')
+
+    if 'encrypted' in volume:
+        return_object['Ebs']['Encrypted'] = volume.get('encrypted')
+
+    return return_object
 
 
 def create_launch_config(connection, module):
     name = module.params.get('name')
-    image_id = module.params.get('image_id')
-    key_name = module.params.get('key_name')
     vpc_id = module.params.get('vpc_id')
     try:
-        security_groups = get_ec2_security_group_ids_from_names(module.params.get('security_groups'), ec2_connect(module), vpc_id=vpc_id, boto3=False)
+        security_groups = get_ec2_security_group_ids_from_names(
+            module.params.get('security_groups'), ec2_connect(module),
+            vpc_id=vpc_id, boto3=False)
```

Comment:
Getting errors setting boto3=True -

The full traceback is:
Traceback (most recent call last):
  File "/var/folders/zy/y1m_yx3n2kg1mxwwshvx_th40000gn/T/ansible_LGvuOa/ansible_module_ec2_lc.py", line 461, in <module>
    main()
  File "/var/folders/zy/y1m_yx3n2kg1mxwwshvx_th40000gn/T/ansible_LGvuOa/ansible_module_ec2_lc.py", line 455, in main
    create_launch_config(connection, module)
  File "/var/folders/zy/y1m_yx3n2kg1mxwwshvx_th40000gn/T/ansible_LGvuOa/ansible_module_ec2_lc.py", line 268, in create_launch_config
    vpc_id=vpc_id, boto3=True)
  File "/var/folders/zy/y1m_yx3n2kg1mxwwshvx_th40000gn/T/ansible_LGvuOa/ansible_modlib.zip/ansible/module_utils/ec2.py", line 540, in get_ec2_security_group_ids_from_names
AttributeError: 'EC2Connection' object has no attribute 'describe_security_groups'

fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "module_stderr": "Traceback (most recent call last):\n  File \"/var/folders/zy/y1m_yx3n2kg1mxwwshvx_th40000gn/T/ansible_LGvuOa/ansible_module_ec2_lc.py\", line 461, in <module>\n    main()\n  File \"/var/folders/zy/y1m_yx3n2kg1mxwwshvx_th40000gn/T/ansible_LGvuOa/ansible_module_ec2_lc.py\", line 455, in main\n    create_launch_config(connection, module)\n  File \"/var/folders/zy/y1m_yx3n2kg1mxwwshvx_th40000gn/T/ansible_LGvuOa/ansible_module_ec2_lc.py\", line 268, in create_launch_config\n    vpc_id=vpc_id, boto3=True)\n  File \"/var/folders/zy/y1m_yx3n2kg1mxwwshvx_th40000gn/T/ansible_LGvuOa/ansible_modlib.zip/ansible/module_utils/ec2.py\", line 540, in get_ec2_security_group_ids_from_names\nAttributeError: 'EC2Connection' object has no attribute 'describe_security_groups'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 0
}

suggestions?